onebusaway-android/src/main/java/com/joulespersecond/seattlebusbot/PreferencesActivity.java

```diff
@@ -149,7 +161,12 @@ public boolean onPreferenceChange(Preference preference, Object newValue) {
             }
         } else if (preference.equals(analyticsPref) && newValue instanceof Boolean) {
             Boolean isAnalyticsActive = (Boolean) newValue;
-            GoogleAnalytics.getInstance(this).setAppOptOut(!isAnalyticsActive);
+            //Report if the analytics turns off, just before shared preference changed
+            if (!isAnalyticsActive)
+                ObaAnalytics.reportEventWithCategory(ObaAnalytics.ObaEventCategory.APP_SETTINGS.toString(),
+                        getString(R.string.analytics_action_edit_general),
+                        getString(R.string.analytics_label_analytic_preference)
+                                + (isAnalyticsActive ? "YES" : "NO"));
```

Comment:
@cagryInside Could you also please add `dispatchLocalHits()` after reporting this event?  I'd like to flush any existing analytics info here, so we don't have analytics continuing to report (e.g., in 30 min) after the user has opted-out.  As we found, I guess this will only work on devices without Google Play Services, but at least its our best effort to be consistent with the user's wishes.  And, if Google Play Services decides to support this in the future, we're already set.